vllm/config.py

```diff
@@ -4188,13 +4195,20 @@ def _set_cudagraph_sizes(self):
             batch_size_capture_list = []
             if self.model_config is not None and \
                 not self.model_config.enforce_eager:
-                batch_size_capture_list = [1, 2, 4
-                                           ] + [i for i in range(8, 513, 8)]
+                cuda_graph_sizes = self.scheduler_config.cuda_graph_sizes
+                if len(cuda_graph_sizes) == 1:
+                    batch_size_capture_list = [1, 2, 4] + [
+                        i for i in range(8, cuda_graph_sizes[0] + 1, 8)
+                    ]
+                elif len(cuda_graph_sizes) > 1:
+                    batch_size_capture_list = sorted(cuda_graph_sizes)
+                else:
+                    raise TypeError(
+                        "cuda_graph_sizes need to be int or list[int]")
```

Comment:
Small nit now that the type has changed:
```suggestion
                        f"Invalid value for {cuda_graph_sizes=}.")
```